src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -268,6 +268,40 @@ def find_manifest_file(deploy_root: Path) -> Path:
     )
 
 
+def find_and_read_manifest_file(deploy_root: Path) -> Any:
+    manifest_file = find_manifest_file(deploy_root=deploy_root)
+    with SecurePath(manifest_file).open(
+        "r", read_file_limit_mb=DEFAULT_SIZE_LIMIT_MB
+    ) as file:
+        manifest_content = safe_load(file.read())
+    return manifest_content
+
+
+def find_setup_script_file(deploy_root: Path) -> Path:
+    """
+    Find the setup script file, if available, in the deploy_root of the Snowflake Native App project.
+    """
+    artifacts = "artifacts"
+    setup_script = "setup_script"
+
+    manifest_content = find_and_read_manifest_file(deploy_root=deploy_root)
+
+    if (artifacts in manifest_content) and (
+        setup_script in manifest_content[artifacts]
+    ):
+        setup_script_rel_path = manifest_content[artifacts][setup_script]
+        # src_path = os.path.realpath(setup_script_rel_path)
+        file_name = Path(deploy_root / setup_script_rel_path)
+        if file_name.exists():
+            return Path(file_name)
+        else:
+            raise SourceNotFoundError(file_name)
```

Comment:
nit: it's not a source, it's the deployed manifest we couldn't find.

Reply:
Updated with the regular `FileNotFoundError`, or do we want ClickException? I like the built in python exc `FileNotFoundError` since a user is much more likely to be familiar with this without reading too much into the error msgs/logs.

Reply:
Doesn't FNFE cause the CLI to crash horribly? I thought ClickException subclasses are handled gracefully and not much else. I was just pointing out that SourceNotFound is used for a particular set of errors, and doesn't seem to match here. The manifest might very well be present, but if the user doesn't have a rule to deploy it, we still won't find it. Maybe we need a new kind of error?